aten/src/ATen/native/native_functions.yaml

```diff
@@ -8142,6 +8142,16 @@
     CPU: searchsorted_cpu
     CUDA: searchsorted_cuda
 
+- func: _convert_coo_to_csr.Tensor(Tensor self, int size, *, bool out_int32=False) -> Tensor
+  dispatch:
+    CPU: _convert_coo_to_csr_cpu
+    CUDA: _convert_coo_to_csr_cuda
+
+- func: _convert_coo_to_csr.Tensor_out(Tensor self, int size, *, bool out_int32=False, Tensor(a!) out) -> Tensor(a!)
```

Comment:
We can improve things here by marking this func with `structured: True` and adding `structured_delegate: _convert_coo_to_csr.out` to `_convert_coo_to_csr`.
Here's more info about structured kernels https://github.com/pytorch/rfcs/blob/rfc-0005/RFC-0005-structured-kernel-definitions.md#structured-keyword-proposal
All error checks and output allocation, that is now duplicated in cpu and cuda paths, would be moved to `TORCH_META_FUNC(_convert_coo_to_csr_out)` (see `aten/src/ATen/TensorMeta.h`).

Reply:
@IvanYashchuk unfortunately structured doesn't currently work with sparse :(

Reply:
I realized now that the name of the function is a bit misleading. It converts the row indices from uncompressed (COO) form to compressed (CSR), probably this function needs renaming. So it operates on strided tensors and structured works here.

Reply:
@ezyang At least when the input is sparse and the output is strided tensor, `structured_delegate` works fine, I use it in my PR https://github.com/pytorch/pytorch/pull/61407
https://github.com/pytorch/pytorch/blob/0788c032bd79f19403e5154540b16ac5c709a7cc/aten/src/ATen/native/native_functions.yaml#L444-L457
TORCH_IMPL_FUNC doesn't work for sparse -- yes, regular c10 dispatched function is used.

`structured_delegate` works because it's allowed currently, reading https://github.com/pytorch/pytorch/blob/master/tools/codegen/gen.py#L173-L177 makes me think it could be changed in the future.

Reply:
Oh ok, dense output does seem like it will work then.